Assets/Script/Gameplay/Player/TrackPlayer.cs

```diff
@@ -123,6 +127,9 @@ public virtual void Initialize(int index, YargPlayer player, SongChart chart, Tr
             var change = ZeroFadePosition - DEFAULT_ZERO_FADE_POS;
             _hudLocation.position = _hudLocation.position.AddZ(change);
 
+            // Must be done after the HUD location is set
+            StarPowerEffect.Initialize();
+
```

Comment:
You should disable the GameObject until it's actually needed, otherwise we're running its Update() loop (and the shader) for no benefit whatsoever most of the time.

```suggestion
            StarPowerEffect.Initialize();
            StarPowerEffect.gameObject.SetActive(false);
 ```
```